backend/src/test/java/com/tradomator/service/WalletServiceTest.java

```diff
@@ -0,0 +1,51 @@
+package com.tradomator.service;
+
+import com.tradomator.model.Balance;
+import com.tradomator.model.Wallet;
+import com.tradomator.repo.WalletRepo;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+class WalletServiceTest {
+
+    WalletRepo walletRepo = mock(WalletRepo.class);
+    WalletService walletService = new WalletService(walletRepo);
+
+    @Test
+    @DisplayName("Test_should_Return_LatestWallet")
+    void getLatestWalletDocumentTest_should_Return_LatestWallet(){
+
+
+        //GIVEN
```

Comment:
No need to define two wallets here. You are telling you're mock which one to return anyways. This test is only for testing that your service calls the correct method in the repo and maps the result to a dto

Reply:
true thanks.